app/main.py

```diff
@@ -1,2 +1,68 @@
+from typing import Hashable, Any
+
+
 class Dictionary:
-    pass
+    LOAD_FACTOR = 2 / 3
+
+    def __init__(self) -> None:
+        self.capacity = 8
+        self.hash_table = [[] for _ in range(self.capacity)]
+        self.threshold = int(self.capacity * self.LOAD_FACTOR)
+        self.size = 0
+
+    def __getitem__(self, key: Hashable) -> Any:
+        hash_key = hash(key)
+        index = hash_key % self.capacity
+
+        while self.hash_table[index]:
+            if (self.hash_table[index][0] == key
+                    and self.hash_table[index][1] == hash_key):
+                return self.hash_table[index][2]
+            index = (index + 1) % self.capacity
+
+        raise KeyError
+
+    def __setitem__(self, key: Hashable, value: Any) -> None:
+        if self.size == self.threshold:
+            self._resize()
+        hash_key = hash(key)
+        index = hash_key % self.capacity
+
+        while True:
+            if not self.hash_table[index]:
+                self.hash_table[index] = [key, hash_key, value]
+                self.size += 1
+                break
+
+            if (self.hash_table[index][0] == key
+                    and self.hash_table[index][1] == hash_key):
+                self.hash_table[index][2] = value
+                break
+
+            index = (index + 1) % self.capacity
+
+    def _resize(self) -> None:
+        new_hash_table = self.hash_table
+        self.size = 0
+        self.capacity *= 2
+        self.hash_table = [[] for _ in range(self.capacity)]
+        self.threshold = int(self.capacity * self.LOAD_FACTOR)
+
+        for elements in new_hash_table:
+            if elements:
+                self.__setitem__(elements[0], elements[2])
+
+    def get(self, key: Hashable, default: Any = None) -> Any:
+        try:
+            return self[key]
+        except KeyError:
+            return default
```

Comment:
why do you need this method? You have a `__getitem__()` method, it is enough

Reply:
Sorry for intrusion, but it's int the extra tasks. :)

>
>  you can implement other methods of the dict interface (both regular and magic):
>   - `clear`
>   - `__delitem__`
>   - `get`
>   - `pop`
>   - `update`
>   - `__iter__`

Reply:
yes, I see now. Thanks!